groups/alteryx-sandbox/nlb.tf

```diff
@@ -0,0 +1,99 @@
+  subnets            = local.alteryx_subnet_id
+
+  tags = {
+    Environment = var.environment
+    Service     = var.application
+  }
+}
+
+# Target configuration for HTTP / port 80
+resource "aws_lb_target_group" "alteryx_sandbox_web" {
+  name        = "${var.application}-${var.application_environment}-web"
+  port        = 80
+  protocol    = "HTTP"
+  target_type = "ip"
+  vpc_id      = data.aws_vpc.vpc.id
+
+  health_check {
+    healthy_threshold   = 2
+    unhealthy_threshold = 2
+    timeout             = "5"
+    path                = "/gallery/api/status/ping/"
+    protocol            = "HTTP"
+    interval            = 30
+    matcher             = "200"
+  }
+}
+
+resource "aws_lb_target_group_attachment" "alteryx_sandbox_web" {
+  target_group_arn = aws_lb_target_group.alteryx_sandbox_web.arn
+  target_id        = module.alteryx_server_ec2[0].private_ip[0]
+  port             = 80
+}
+
+# Configuration for Certificate
+resource "aws_acm_certificate" "certificate" {
+  count = local.create_ssl_certificate ? 1 : 0
+
+  domain_name               = "${var.application}.${var.environment}.${local.dns_zone_name}"
+  subject_alternative_names = ["*.${var.application}.${var.environment}.${local.dns_zone_name}"]
+  validation_method         = "DNS"
+}
+
+resource "aws_route53_record" "certificate_validation" {
+  for_each = local.create_ssl_certificate ? {
+    for dvo in aws_acm_certificate.certificate[0].domain_validation_options : dvo.domain_name => {
+      name   = dvo.resource_record_name
+      type   = dvo.resource_record_type
+      record = dvo.resource_record_value
+    }
+  } : {}
+
+  allow_overwrite = true
+  name            = each.value.name
+  records         = [each.value.record]
+  ttl             = 60
+  type            = each.value.type
+  zone_id         = local.dns_zone_id
+}
+
+resource "aws_acm_certificate_validation" "certificate" {
+  count = local.create_ssl_certificate ? 1 : 0
+
+  certificate_arn         = aws_acm_certificate.certificate[0].arn
+  validation_record_fqdns = [for record in aws_route53_record.certificate_validation : record.fqdn]
+}
+
+# Listener configuration
+resource "aws_lb_listener" "alteryx_sandbox_listener_443" {
+  load_balancer_arn = aws_lb.alteryx_sandbox.arn
+  port              = "443"
+  protocol          = "HTTPS"
+  certificate_arn   = local.ssl_certificate_arn
+  ssl_policy        = "ELBSecurityPolicy-TLS13-1-2-2021-06"
+  default_action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.alteryx_sandbox_web.arn
+  }
+}
+
+# Create a listener resource to redirect HTTP to HTTPS
+resource "aws_lb_listener" "alteryx_sandbox_listener_80_redirect" {
+  load_balancer_arn = aws_lb.alteryx_sandbox.arn
+  port              = "80"
+  protocol          = "HTTP"
+  default_action {
+    type = "redirect"
+    redirect {
+      port        = "443"
+      protocol    = "HTTPS"
+      status_code = "HTTP_301"
+    }
+  }
+}
```

Comment:
Given that we're switching to a layer-4 (network) load balancer, the protocol for this listener will need to be updated to `TCP`.

In addition, the HTTPS upgrade can't be handled in this way as we're operating at layer 4 of the OSI model when using a network load balancer. Likely the best option in this scenario is to introduce a layer-7 (application) load balancer, and forward traffic for this listener to the ALB, which can then handle the upgrade.

See [Application Load Balancers as targets](https://docs.aws.amazon.com/elasticloadbalancing/latest/network/application-load-balancer-target.html) for more information, but in summary:

> You can create a target group with a single Application Load Balancer as the target, and configure your Network Load Balancer to forward traffic to it.